src/garage/torch/distributions/tanh_normal.py

```diff
@@ -0,0 +1,262 @@
+"""A Gaussian distribution with tanh transformation."""
+import torch
+from torch.distributions import Normal
+from torch.distributions.independent import Independent
+
+
+class TanhNormal(torch.distributions.Distribution):
+    r"""A distribution induced by applying a tanh transformation to a Gaussian random variable.
+
+    Algorithms like SAC and Pearl use this transformed distribution.
+    It can be thought of as a distribution of X where
+        :math: `Y ~ \mathcal{N}(\mu, \sigma)`
+        :math: `X = tanh(Y)`
+
+    Args:
+        loc (torch.Tensor): The mean of this distribution.
+        scale (torch.Tensor): The stdev of this distribution.
+
+    """ # noqa: 501
+
+    def __init__(self, loc, scale):
+        self._normal = Independent(Normal(loc, scale), 1)
+        super().__init__()
+
+    def log_prob(self, value, pre_tanh_value=None, epsilon=1e-6):
+        """The log likelihood of a sample on the this Tanh Distribution.
+
+        Args:
+            value (torch.Tensor): The sample whose loglikelihood is being
+                computed.
+            pre_tanh_value (torch.Tensor): The value prior to having the tanh
+                function applied to it but after it has been sampled from the
+                normal distribution.
+            epsilon (float): Stabilization coefficient.
```

Comment:
Regularization constant. Making this value larger (smaller?) makes the computation more stable but less precise.

Reply:
larger